pkg/virtual/framework/dynamic/types.go

```diff
@@ -31,32 +29,13 @@ var _ framework.VirtualWorkspace = (*DynamicVirtualWorkspace)(nil)
 // DynamicVirtualWorkspace is an implementation of a framework.VirtualWorkspace which can dynamically serve resources,
 // based on API definitions (including an OpenAPI v3 schema), and a Rest storage provider.
 type DynamicVirtualWorkspace struct {
-	Name             string
-	RootPathResolver framework.RootPathResolverFunc
-	Authorizer       authorizer.AuthorizerFunc
-	Ready            framework.ReadyFunc
+	framework.Namer
+	framework.RootPathResolver
+	authorizer.Authorizer
+	framework.ReadyChecker
 
 	// BootstrapAPISetManagement creates, initializes and returns an apidefinition.APIDefinitionSetGetter.
 	// Usually it would also set up some logic that will call the apiserver.CreateServingInfoFor() method
 	// to add an apidefinition.APIDefinition in the apidefinition.APIDefinitionSetGetter on some event.
 	BootstrapAPISetManagement func(mainConfig genericapiserver.CompletedConfig) (apidefinition.APIDefinitionSetGetter, error)
```

Comment:
I would just as soon make this a typed function as well?

Reply:
We didn't have any shared implementations for this that we wanted to re-use between workspaces, so I wasn't keen to tackle it for now.